### DISCUSSION TITLE
Request for small screen support

### DISCUSSION THREAD

**Discussion by bsaunder2002** [jithware/brethap#89] [Ideas]
I use a  Full Android Watch (FAW).  This is NOT a WearOS, proprietary OS, or dumb smartphone.  It's a full Android 10 with F-Droid and Play store, 4G, GPS, ...  All for $118.

Since it's a watch, it has a tiny screen.

If you decided on supporting this idea, think tiny with swipe actions over pressing little icons.
Smartwatches come in squares and circles.  Mine is a circle so the corners are hidden.  If an app doesnt place items in the circle, they are not seen.
(Interestingly, I can run droidVNC-NG which is a Android VNC server on the watch.  It shows there is a square screen...just chopped off.  Sorry I do not know the size.)

Brethap works on the watch as is but here are some items.
Swipe right to bring up menu though About Brethap is "out Brethap".
The preference screen is badly chopped off.  From the watch I cannot see/click the vertical ...'s So I cannot change to the presets.
From the main screen, again chopped off, I see "-8 Breathing", Press Start, the blue breathing icon, the time at the bottom, and unfortunately, the tiniest sliver of the green > go button.  It's 1 time out of 10 attempts clickable.

Some ideas:
Center the name of the current pattern.
Option to enlarge time font...but how to decide where to place it...I dont know.
Center the green > where the blue breath circle will appear.  Once clicked the green goes away and if you tap the active blue it becomes a blue with the pause button.
Or...always a tiny blue button and right above/below the time say "tap to start or pause"

I open you consider making BrethAp functional on small screens.  I often take my watch running and leave everything else at home.  I'd like BrethAp to have a cool down pattern.

Awesome product.

**Comment by jithware** [comment]
Brethap is only supported for:

- Watches running Wear OS, see [brethap_wear](https://github.com/jithware/brethap_wear#brethap-wear)
- Android phones, tablets and chromebooks, see [brethap](https://github.com/jithware/brethap#brethap)
- Web browsers, see [web](https://jithware.github.io/brethap)

Unfortunately, it is not feasible to support watches with the brethap "phone" [code base](https://github.com/jithware/brethap).

**Reply by jithware** [reply]
With the above being said, since Brethap is open source, there is nothing stopping one from building it yourself. Not familar with FAWs, but if you can manually install apks, you could try to build brethap_wear and install it on your FAW. First install [flutter](https://docs.flutter.dev/get-started/install) and [git](https://github.com/git-guides/install-git) then:

```
git clone https://github.com/jithware/brethap_wear.git
cd brethap_wear/
flutter build apk --split-per-abi --debug
```

For more info on building apks see [build-an-apk](https://docs.flutter.dev/deployment/android#build-an-apk).

**Comment by bsaunder2002** [comment]
I will try Play / Brethap_wear.  I try to stay on F-droid which (I dont think) shows Wear OS apps.  I'll try Play.

Also, I think I can hack enough to make those changes to the Brethap phone.  I'll send an ignorable pull request with my changes if I get them done. :) 

(FAW is just a term describing full Android on a watch.  My $100 even has 4G phone built in.  Crazy. )


Again THANKS for Brethap...it is helping many people lead healthier (pubmed breathing) calmer lives (pubmed breathing anxiety).!
